scripts/lib/setup_venv.py

```diff
@@ -269,7 +269,13 @@ def do_setup_virtualenv(venv_path, requirements_file, virtualenv_args):
     if not try_to_copy_venv(venv_path, new_packages):
         # Create new virtualenv.
         run(["sudo", "mkdir", "-p", venv_path])
-        run(["sudo", "virtualenv"] + virtualenv_args + [venv_path])
+        proxy_env = [
+            "env",
+            "http_proxy=" + os.environ.get("http_proxy", ""),
+            "https_proxy=" + os.environ.get("https_proxy", ""),
+            "no_proxy=" + os.environ.get("no_proxy", ""),
+        ]
+        run(["sudo", "-H"] + proxy_env + ["virtualenv"] + virtualenv_args + [venv_path])
```

Comment:
Have you tested this?  I think you want to be modifying `install_venv_deps`, not `virtualenv`, since that doesn't download anything.  

Also, is adding the `-H` argument required?  It shouldn't be.

Reply:
Sorry it took so long to respond. I have tested it and I don't need the proxy settings for this step.
Also, the downloading of the deps seem to work, probably because pip does use the `http_proxy` settings (I've passed them using the `--build-args` on the docker build command).

Reply:
I've updated the request to only include the change to the provision step.